### PR TITLE
Pixeltime AOV

### LINKED ISSUE #333: Add a debug AOV to have render time per pixel
When you have big render times, it can be very interesting to have a debug AOV showing where the renderer is spending it's time.

To give you an overview, this render:

Provide a such "time per pixel" buffer:

(Images from a personal project from [Tristan North](http://www.imdb.com/name/nm2348845/), thanks to him!)

More it's bright, more it takes time.

There is may way to display time per pixel:
- A 0-1 range frame with reel range values stored in metadatas (and log if the final image doesn't support metadatas. This bring to the most visual feedback but it's difficult, when you put this image to Nuke, to get a pixel value and exactly know which time it represent. While visually immediate, this is not my personal taste.
- A 0-n range, where N is the pixel time in second.I think this is the more efficient because you can change the exposure value in Nuke and keep the reel pixel value available when you click on it. It's also the simpler to implement as you don't have to deal with a "final range" that imply to have the image finished.
### References
- Mental ray's [time per pixel diagnostic](https://elementalray.wordpress.com/2012/01/29/unified-sampling-redux/).
- It's called [cputime](https://support.solidangle.com/display/AFMUG/AOVs) in Arnold.

### PR DESCRIPTION
**Pull request by est77**
- Add a PixelTime AOV (useful for debugging).
- Switch DefaultProcessorTimer on Linux and FreeBSD to use clock_gettime()

The PR also includes a quick PNG writting fix for 1 & 2 channel images to avoid appleseed crashing.
The good fix is in #1803, when that PR is ready to be merged.

Closes #333

### CODE REVIEW

File: src/appleseed/renderer/modeling/aov/diagnosticaov.cpp [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+        auto_release_ptr<AOVAccumulator> create_accumulator() const override
+        {
+            return auto_release_ptr<AOVAccumulator>(new PixelTimeAOVAccumulator(get_image()));
+        }
+    };
+}
+
+
+//
+// DepthAOVFactory class implementation.
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/aov/diagnosticaov.cpp]: Fix comment.

File: src/appleseed/renderer/modeling/aov/diagnosticaov.cpp [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    return Model;
+}
+
+Dictionary PixelTimeAOVFactory::get_model_metadata() const
+{
+    return
+        Dictionary()
+            .insert("name", Model)
+            .insert("label", "Pixel Time")
+            .insert("default_model", "true");
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/aov/diagnosticaov.cpp]: Depth AOV is already the default model.

**Comment by est77** [comment]
> Maybe rename diagnosticaov.{h,cpp} to pixeltimeaov.{h,cpp}?

I was planning to add some more smaller diagnostic AOVs into this file:
Invalid pixels, max ray depth and maybe some other.

I think that they are small enough so that they could go into only one file 
and maybe share some code.

**Comment by est77** [comment]
Fixed and rebased

**Comment by dictoon** [comment]
> I was planning to add some more smaller diagnostic AOVs into this file: Invalid pixels, max ray depth and maybe some other.

In that case, maybe `diagnosticaovs.{h,cpp}`?

> I think that they are small enough so that they could go into only one file and maybe share some code.

Are they really smaller than the others? I'm fine either way (single or multiple pair of files).

File: src/appleseed/foundation/platform/defaulttimers.cpp [hunk 3]
```diff
@@ -40,7 +40,10 @@
 #if defined _WIN32
 #include <sys/timeb.h>
 #include <sys/types.h>
-#elif defined __GNUC__
+#elif defined __linux__ || defined __FreeBSD__
+#include<time.h>
```
Comment (dictoon) [inline src/appleseed/foundation/platform/defaulttimers.cpp]: Add space before `<`

File: src/appleseed/renderer/modeling/aov/pixeltimeaov.cpp [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+#include "foundation/utility/api/specializedapiarrays.h"
+#include "foundation/utility/containers/dictionary.h"
+#include "foundation/utility/stopwatch.h"
+
+// Standard headers.
+#include <algorithm>
+#include <cstddef>
+#include <vector>
+
+#include<time.h>
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/aov/pixeltimeaov.cpp]: Add space before `<`.

Also, maybe `<ctime>`?

**Comment by est77** [comment]
Fixed

**Pull request merged**